extensions/dicom-seg/src/components/SegmentationPanel.js

```diff
@@ -0,0 +1,473 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import cornerstoneTools from 'cornerstone-tools';
+import cornerstone from 'cornerstone-core';
+import moment from 'moment';
+
+import { utils } from '@ohif/core';
+import { ScrollableArea, TableList, Icon } from '@ohif/ui';
+
+import {
+  BrushColorSelector,
+  BrushRadius,
+  SegmentationItem,
+  SegmentItem,
+  SegmentationSelect,
+} from './index';
+
+import './SegmentationPanel.css';
+import SegmentationSettings from './SegmentationSettings';
+
+const { studyMetadataManager } = utils;
+
+/* TODO: wrap get/set interactions with the cornerstoneTools
+ * store with context to make these kind of things less blurry. */
+const segmentationModule = cornerstoneTools.getModule('segmentation');
+const { configuration } = segmentationModule;
+const DEFAULT_BRUSH_RADIUS = configuration.radius || 10;
+
+const refreshViewport = () => {
+  cornerstone.getEnabledElements().forEach(enabledElement => {
+    cornerstone.updateImage(enabledElement.element);
```

Comment:
Slight timing thing here to be aware of. This may blow up if the enabledElement does not yet have a displayed image.

Reply:
@dannyrb what type of check i can do here in this case?